plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryInstanceCleaner.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.bigquery;
+
+import com.google.cloud.bigquery.TableResult;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import io.airlift.log.Logger;
+import io.trino.plugin.bigquery.BigQueryQueryRunner.BigQuerySqlExecutor;
+import io.trino.tpch.TpchTable;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map.Entry;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.bigquery.BigQueryQueryRunner.TEST_SCHEMA;
+import static io.trino.plugin.bigquery.BigQueryQueryRunner.TPCH_SCHEMA;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toUnmodifiableSet;
+
+public class TestBigQueryInstanceCleaner
+{
+    public static final Logger LOG = Logger.get(TestBigQueryInstanceCleaner.class);
+
+    /**
+     * List of table names that will not be dropped.
+     */
+    private static final Collection<String> tablesToKeep = TpchTable.getTables().stream()
+            .map(TpchTable::getTableName)
+            .map(s -> s.toLowerCase(Locale.ENGLISH))
+            .collect(toUnmodifiableSet());
+
+    // see https://cloud.google.com/bigquery/docs/information-schema-tables#tables_view for possible values
+    public static final Collection<String> tableTypesToDrop = ImmutableList.of("BASE TABLE", "VIEW", "MATERIALIZED VIEW");
+
+    private BigQuerySqlExecutor bigQuerySqlExecutor;
+
+    @BeforeClass
+    public void setUp()
+    {
+        this.bigQuerySqlExecutor = new BigQuerySqlExecutor();
+    }
+
+    @Test(dataProvider = "cleanUpSchemasDataProvider")
+    public void cleanUpTables(String schemaName)
+    {
+        logObjectsCount(schemaName);
+        if (!tablesToKeep.isEmpty()) {
+            LOG.info("Will not drop these tables: %s", join(", ", tablesToKeep));
+        }
+
+        dropTablesCreatedBeforeYesterday(schemaName);
+        logObjectsCount(schemaName);
+    }
+
+    @DataProvider
+    public static Object[][] cleanUpSchemasDataProvider()
```

Comment:
Do we have any dynamically created schemas?

Reply:
Yes, but those get dropped completely via `cleanUpDatasets` (in this same class).
We add a label to all datasets created through the tests and that method drops all of labelled ones that are older than a day.

Reply:
Add a  comment here saying that other schemas, test-created schemas, are taken care elsewhere.